com.wamas.ide.launching/src/com/wamas/ide/launching/validation/LcDslValidator.xtend

```diff
@@ -366,12 +367,20 @@ class LcDslValidator extends AbstractLcDslValidator {
 		val type = javaProject.findType(className)
 		if (type === null) {
 			error("Test class " + className + " must point to an existing class in project " + javaProject, cfg.test, LC.testConfig_Class)
-		} else if (!Flags.isPublic(type.flags)) {
-			error("Test class " + className + " must be public.", cfg.test, LC.testConfig_Class)
-		} else if (type.methods.stream().filter([method | Flags.isPublic(method.flags)]).flatMap([method | method.annotations.stream]).noneMatch([annotation | TEST_ANNOTATION_NAMES.contains(annotation.elementName)])) {
-			error("Test class " + className + " must at least have one method annotated with one annotation out of " + TEST_ANNOTATION_NAMES, cfg.test, LC.testConfig_Class)
+		} else if (Flags.isPrivate(type.flags)) {
+			error("Test class " + className + " must not be private.", cfg.test, LC.testConfig_Class)
+		} else if (!hasTestMethod(type) && !hasNestedTest(type)) {
+			error("Test class " + className + " must at least have one method annotated with one annotation out of " + TEST_ANNOTATION_NAMES + " or must contain a nested test.", cfg.test, LC.testConfig_Class)
 		}
 	}
+	
+	private def boolean hasTestMethod(IType type) {
+		return type.methods.stream().filter([method | Flags.isPublic(method.flags)]).flatMap([method | method.annotations.stream]).anyMatch([annotation | TEST_ANNOTATION_NAMES.contains(annotation.elementName)])		
```

Comment:
Test methods do not need to be public either, so the filter should be `!Flags.isPrivate(method.flags)` as well.